### PR TITLE
[Tensorflow] update typing extensions for tf-2.4 training

### LINKED ISSUE #1393: [sanity] fix typing-extensions version for TF2.4 Training DLCs
## Checklist
- [ ] I've prepended issue tag with type of change: [feature]
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [x] (If applicable) I've documented below the test files this relates to

#### Concise Description:
TF 2.4.3 sanity check is failing due to `pip check` failing. The failure is related to conflict in dependency package where 

>tensorflow-cpu 2.4.3 has requirement typing-extensions~=3.7.4, but you have typing-extensions 3.10.0.2.

Investigation found that the dependency chain of `typing-extensions` is below:
```
typing-extensions==3.10.0.0
  - importlib-metadata==4.5.0 [requires: typing-extensions>=3.6.4]
    - pluggy==0.13.1 [requires: importlib-metadata>=0.12]
      - pytest==6.2.4 [requires: pluggy>=0.12,<1.0.0a1]
        - dlc-release==0.3.0 [requires: pytest]
    - pytest==6.2.4 [requires: importlib-metadata>=0.12]
      - dlc-release==0.3.0 [requires: pytest]
    - virtualenv==20.4.7 [requires: importlib-metadata>=0.12]
      - pipenv==2020.11.15 [requires: virtualenv]
```
And we can see from above dependency chain, there is no explicit requirement for `typing-extensions~=3.7.4`. Which was why `typing-extensions>=3.6.4` was used by pip and installed version `3.10.0.0`.

We'll need to update the dockerfiles and explicitly install `typing-extensions~=3.7.4`.

#### DLC image/dockerfile:
- [ ] https://github.com/aws/deep-learning-containers/blob/f54ff8ea74f2a6a1d3d63745688fc5c2f60491ed/tensorflow/training/docker/2.4/py3/Dockerfile.cpu#L148
- [ ] https://github.com/aws/deep-learning-containers/blob/f54ff8ea74f2a6a1d3d63745688fc5c2f60491ed/tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu#L230

#### Additional context

Related test: https://github.com/aws/deep-learning-containers/blob/f54ff8ea74f2a6a1d3d63745688fc5c2f60491ed/test/dlc_tests/sanity/test_pre_release.py#L395

### PR DESCRIPTION
**Pull request by junpuf**
fixes #1393

### Description

Explicitly install "typing-extensions~=3.7.4".

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron"/"graviton">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by arjkesh** [comment]
Did you identify where in the dependency tree this version conflict was coming from?

**Comment by arjkesh** [comment]
Also, you'll need to update `buildspec.yml` in order to build the TF 2.4 images in PR

**Comment by junpuf** [comment]
> Did you identify where in the dependency tree this version conflict was coming from?

Yes, see #1393 for detail

**Comment by junpuf** [comment]
> Also, you'll need to update `buildspec.yml` in order to build the TF 2.4 images in PR

Right, let me copy the old buildspec here. Also, i notices that TF1 images gets build, is it configurable from developer_config.toml?

**Comment by arjkesh** [comment]
> > Also, you'll need to update `buildspec.yml` in order to build the TF 2.4 images in PR
> 
> Right, let me copy the old buildspec here. Also, i notices that TF1 images gets build, is it configurable from developer_config.toml?

We don't have this option at the moment, but I thought it was such a good idea that I made a quick [PR](https://github.com/aws/deep-learning-containers/pull/1403) for it

**Comment by tejaschumbalkar** [comment]
The sanity test failure is now fixed. PR is good to merge.

**Review by tejaschumbalkar — APPROVED** [review]

**Comment by satishpasumarthi** [comment]
LGTM
